File: 프로그래머스/0/120921. 문자열 밀기/문자열 밀기.js
function solution(A, B) {
    let arr = [...A]
    for (let i = 0; i < arr.length; i++) {
        if (A === B) {
            return i;
        } else {
            arr.unshift(arr.pop());
            if (arr.join('') === B) {
                return i + 1;
            }
        }
    }
    return -1;
}

File: 프로그래머스/0/120921. 문자열 밀기/문자열 밀기.py
def solution(A, B):
    if A == B:
        return 0
    n = len(A)
    current_A = list(A)
    for i in range(1, n + 1):
        last_char = current_A.pop()
        current_A.insert(0, last_char)
        
        if "".join(current_A) == B:
            return i
    return -1